### PR TITLE
Ban assembly versions with wildcards in deterministic builds

### LINKED ISSUE #4578: Compilation of source with auto-generated version is non-deterministic
```C#
[assembly: System.Reflection.AssemblyVersion("1.0.*")]
```

The compiler generates the version based on the current time:

```C#
// src\Compilers\Core\Portable\VersionHelper.cs:
...
            if (hasWildcard)
            {
                int seconds = ((int)(DateTime.Now.TimeOfDay.TotalSeconds)) / 2;
                if (seconds > (int)maxValue)
                {
                    version = null;
                    return false;
                }
                values[3] = (ushort)seconds;

                if (elements.Length == 3)
                {
                    TimeSpan days = DateTime.Today - new DateTime(2000, 1, 1);
...
```

### PR DESCRIPTION
**Pull request by TyOverby**
Closes #4578

### CODE REVIEW

**Comment by TyOverby** [comment]
review: @gafter @tmat @jaredpar

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs [hunk 1]
```diff
@@ -2073,7 +2073,8 @@ private void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArguments<Attr
             {
                 string verString = (string)attribute.CommonConstructorArguments[0].Value;
                 Version version;
-                if (!VersionHelper.TryParseAssemblyVersion(verString, allowWildcard: true, version: out version))
+                bool deterministic = _compilation.Feature("deterministic")?.Equals("true", StringComparison.OrdinalIgnoreCase) ?? false;
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs]: @gafter do we have a standard way of checking for deterministic compilation that we could reuse here?

**Comment by jaredpar** [comment]
:+1:

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs (hunk 1 above)
Comment (TyOverby) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs]: Doh

`compilation.IsEmitDeterministic` exists.  Don't see how that one slipped by me.

Comment (TyOverby) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs]: Fixed in 821de3c

**Comment by jaredpar** [comment]
test mac please

**Comment by jaredpar** [comment]
retest mac please

**Comment by jaredpar** [comment]
test mac please

**Comment by tannergooding** [comment]
test mac please

**Comment by tannergooding** [comment]
test mac please

**Comment by TyOverby** [comment]
@jaredpar @tannergooding 

How do I merge this?

**Pull request merged**